### ISSUE TITLE
repr.plot.res does not work as expected

### ISSUE THREAD

**Issue by dfrankow** [IRkernel/IRkernel#458]
I found repr.plot defaults [here](https://github.com/IRkernel/repr/blob/master/R/options.r#L51-L61), including res, which looks to be dots per inch (dpi). However, when I raise the res from 120 to 300, the plot gets bigger, despite me setting height and width, instead of staying the same size and getting crisper, as I would hope.

========================================

More detail from [my stack overflow question](http://stackoverflow.com/questions/43123034/how-to-get-high-resolution-plots-in-a-jupyter-notebook-using-the-r-kernel):

Using SVG in a Jupyter notebook with R kernel mangles the text. See [here](http://compbio.ucsd.edu/outputting-beautiful-jupyter-notebooks-r-kernel-edition/#html-output-mangled-graphics-text) and [here](https://github.com/jupyter/nbconvert/issues/129). The workaround is to use PNG. It's a bit blurry. In Python, you can ask the PNG renderer to use more dots per inch with

```InlineBackend.figure_format = 'retina'```

See for example [trick 15 here](https://www.dataquest.io/blog/jupyter-notebook-tips-tricks-shortcuts/).

How can I get higher resolution plots using the R kernel? SVG workaround? PNG renderer settings?

I'm using ggplot in R.

**Comment by flying-sheep** [comment]
> Using SVG in a Jupyter notebook with R kernel mangles the text

this was a very old bug, which doesn’t exist anymore. SVGs should be isolated in an iframe which prevents the text issues.

> However, when I raise the res from 120 to 300, the plot gets bigger, despite me setting height and width, instead of staying the same size and getting crisper, as I would hope.

yes. `width` and `height` are in inches, and res defines how many `ppi` are used, so the plots will be of dimensions width·res × height·res.

[this](https://github.com/IRkernel/IRkernel/blob/d16c2e9cecfcd4a2c74952294c57e9f8ee4aab24/R/execution.r#L217-L218) would be the right point to hook in. when a specific option is set, we just divide those numbers by 2.

**Comment by dfrankow** [comment]
1. **hi-res option**. Dividing by (repr.plot.res / 120) sounds good to me.  The fact that `res` is described as "dots per inch" makes me think it won't change plot size, just crispness.

Unfortunately, I'm a long way from being able to set up a test IRKernel and trying a fix.

2. **SVG bug.**

When you say "this was a very old bug", it's hard for me to tell exactly how to get the fix. I've reproduced it in our current spark notebook setup. That starts from a Dockerfile like this:

```FROM jupyter/all-spark-notebook:749017d878d6```

That tag is [here](https://hub.docker.com/r/jupyter/all-spark-notebook/tags/) and says "7 months ago" (as of April 2017), so it's not too terribly old. Upgrading to the latest all-spark-notebook causes other problems, so it would be useful to know exactly which release the bugfix is in.

Moreover, [this nbconvert bug](https://github.com/jupyter/nbconvert/issues/129) is still open, so it's not clear to me it's even fixed in the right place.

**Comment by flying-sheep** [comment]
> Unfortunately, I'm a long way from being able to set up a test IRKernel and trying a fix.

did you follow [the install instructions](https://irkernel.github.io/installation/)? then you can just

1. start a notebook server using `jupyter notebook`
2. notebook: open/create an R notebook
3. open the IRkernel project with RStudio
4. rstudio: make changes
5. rstudio: hit [“Build & Reload”](http://www.rstudio.com/images/docs/build_pane.png)
6. notebook: click “Kernel → restart & run all cells”
7. if it didn’t work goto 4

> When you say "this was a very old bug", it's hard for me to tell exactly how to get the fix.

sorry: long ago, it was a bug in one of the IRkernel organization packages, but i can’t remember which one. so the fix is to upgrade IRkernel, IRdisplay, and repr to the newest CRAN (IRdisplay and repr) or git versions (IRkernel, as it isn’t on CRAN yet)

> Moreover, this nbconvert bug is still open, so it's not clear to me it's even fixed in the right place.

that one is about nbconvert’s HTML conversion. i assumed “Using SVG in a Jupyter notebook with R kernel mangles the text” meant that you’re using the interactive notebook environment, where an iframe is used, preventing this from happening.

**Comment by raivivek** [comment]
@flying-sheep I attempted a fix at this. It appears to work.

**Comment by Youguang** [comment]
> this was a very old bug, which doesn’t exist anymore. SVGs should be isolated in an iframe which prevents the text issues.

Multi SVGs show well in Jupyter-notebook, but mess when output as html.
```
Jupyter-notebook --version 5.7.6
R version 3.5.2
IRkernel_0.8.15 IRdisplay_0.7.0 repr_0.19.2  
```

> either we add a “hi-res” option and just divide those numbers by 2 when it’s set. or we derive the factor from the `res` value. (120 → ÷1; 240 → ÷2; …)

I am still not clear about how to get high resolution inline png. I am a novice, could you explain it in detail? Thank you.

**Comment by flying-sheep** [comment]
Well, @raivivek didn’t respond to my last message in #606, so there’s no fix yet. @raivivek: are you still interested in seeing this through? Else I’m going to finish it.

**Comment by flying-sheep** [comment]
Should be fixed in 44a92feba841e9959788f3c6750c73d293ca9387 (v1.1), if not please say so.
